source/AI.cpp

```diff
@@ -812,7 +814,7 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			// Miners with free cargo space and available mining time should mine. Mission NPCs
 			// should mine even if there are other miners or they have been mining a while.
 			if(it->Cargo().Free() >= 5 && IsArmed(*it) && (it->IsSpecial()
-					|| (++miningTime[&*it] < 3600 && ++minerCount < maxMinerCount)))
+					|| (++miningTime[&*it] < GameData::GetGamerules().NPCMaxMiningTime() && ++minerCount < maxMinerCount)))
```

Comment:
This line is really long and hard to read.
Create a `const int npcMaxMiningTime = GameData::GetGamerules().NPCMaxMiningTime();` probably outside this big for loop, and just refer to that here.